### PR TITLE
Add cancel, back, and remove flows for array builder

### PR DESCRIPTION
**Pull request by rhasselle-oddball**
## Summary

- Add cancel, back, and remove flows for array builder
- Includes modals and tests

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team-forms#903

## Testing done

Browser and Cypress tests

## Screenshots

image
image
image
image

## What areas of the site does it impact?

New array builder pattern

### Quality Assurance & Testing

  [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/simple-forms/mock-simple-forms-patterns/config/form.js [hunk 1]
```diff
@@ -253,6 +260,13 @@ const formConfig = {
           arrayPath: 'employers',
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/mock-simple-forms-patterns/config/form.js:260]: if you look at the overall way the array pages are defined here, it's still pretty verbose, but at least now we have options for everything. 
- onNavBack/onNavForward: we need to control what happens on back and on continue (for example we might need to remove an item on pressing back, or keeping the add=true urlParams when going to the next page)
- ContentBeforeButtons for adding a cancel button that changes routes after a modal

goal will be to simplify this further once everything's established

File: src/applications/simple-forms/mock-simple-forms-patterns/pages/mockArrayMultiPageBuilderItemPage1.js [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
   addressNoMilitarySchema,
   addressNoMilitaryUI,
   titleUI,
 } from 'platform/forms-system/src/js/web-component-patterns';
 import { VaTextInputField } from 'platform/forms-system/src/js/web-component-fields';
+import ArrayBuilderCancelAddingButton from '../arrayBuilder/components/ArrayBuilderCancelAddingButton';
 
 /** @type {PageSchema} */
 export default {
+  ContentBeforeButtons: ({ router }) => (
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/mock-simple-forms-patterns/pages/mockArrayMultiPageBuilderItemPage1.js:12]: these options might be able to be reduced to `noun` and `pluralNoun` if we want to

File: src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/components/ArrayBuilderCancelAddingButton.jsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
+/**
```
Comment (rhasselle-oddball) [inline src/applications/simple-forms/mock-simple-forms-patterns/arrayBuilder/components/ArrayBuilderCancelAddingButton.jsx:1]: these 'ArrayBuilder' specific components and helpers will eventually be moved to forms-system once we're happy with the pattern after we've built it all

**Review by rhasselle-oddball — COMMENTED** [review]
dev notes

**Review by ndsprinkle — APPROVED** [review]

**Pull request merged**